core/arch/mangle_shared.c

```diff
@@ -1824,7 +1824,8 @@ find_syscall_num(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr)
 
     if (prev == NULL) {
 #if defined(WINDOWS) && defined(X64)
-        if (get_os_version() >= WINDOWS_VERSION_10_1511) {
+        if (get_os_version() >= WINDOWS_VERSION_10_1511 &&
+            !is_wow64_process(NT_CURRENT_PROCESS)) {
```

Comment:
Repeating comment from prior review: when in 64-bit mode we still want this, so we want to check the current mode: since it's passed instrs, the instr mode makes the most sense: `instr_get_isa_mode`.